src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -384,18 +387,23 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
                             let zAxisMultiplier = this._scene.useRightHandedSystem? 1 : -1;
                             const indexTipPos = indexTipPose.transform.position;
                             const indexTipOrientation = indexTipPose.transform.orientation;
+                            controllerData.indexTipQuaternion.set(indexTipOrientation.x, indexTipOrientation.y, (indexTipOrientation.z * zAxisMultiplier), (indexTipOrientation.w * zAxisMultiplier));
+
                             // set positions for near pick and hover
                             if (controllerData.pickIndexMeshTip) {
                                 controllerData.pickIndexMeshTip.position.set(indexTipPos.x, indexTipPos.y, (indexTipPos.z * zAxisMultiplier));
                             }
                             if (controllerData.hoverIndexMeshTip) {
-                                controllerData.hoverIndexMeshTip.position.set(indexTipPos.x, indexTipPos.y, (indexTipPos.z * zAxisMultiplier));
+                                controllerData.hoverIndexMeshTip.position.set(indexTipPos.x, indexTipPos.y, indexTipPos.z);
                             }
+
                             // set near interaction grab ray parameters
                             let nearGrabRayLength = 0.5;
-                            let indexTipZaxisMultiplier = -1;
+                            let webxrZaxisMultiplier = -1;
+                            let indexTipOrientationVector = Vector3.Zero();
                             controllerData.grabRay.origin.set(indexTipPos.x, indexTipPos.y, (indexTipPos.z * zAxisMultiplier));
-                            controllerData.grabRay.direction.set(indexTipOrientation.x, indexTipOrientation.y, (indexTipOrientation.z) * indexTipZaxisMultiplier);
+                            controllerData.indexTipQuaternion.toEulerAnglesToRef(indexTipOrientationVector);
+                            controllerData.grabRay.direction.set(indexTipOrientationVector.x, indexTipOrientationVector.y, (indexTipOrientationVector.z) * webxrZaxisMultiplier);

```

Comment:
This seems to apply a negation on top of the negation applied earlier, cancelling it out?

Reply:
let me check

Reply:
remember the ray projected from the index tip is in the opposite direction, that is why we negate it here. i dont think it has to do with Right handed system here. that is why i hardcoded it